rozy/src/main.rs

```diff
@@ -256,33 +270,14 @@ fn update(path_to_ozy: &std::path::PathBuf, url: &Option<String>) -> Result<()>
         perms.set_mode(0o755);
         std::fs::set_permissions(&dest_path, perms)?;
 
-        let mut command = exec::Command::new(&dest_path);
-        command.arg("update");
-
-        // We use exec to replace this process with the child entirely. The only time exec() returns is
-        // if there was an error.
-        let err = command.exec();
+        let err = exec::execvp(&dest_path, std::env::args().collect::<Vec<String>>());
```

Comment:
The key trick here is that we execute the new binary with the exact same argv as ourselves, which means it will end up back in `update`, _and_ eventually execute the real app if we were invoked that way.